### PR TITLE
Bump kafka-journal and other dependencies, migrate to new RebalanceCallback1

### PR DESCRIPTION
**Pull request by Z1kkurat**

### CODE REVIEW

**Comment by Z1kkurat** [comment]
Latest build failure caused by https://github.com/embeddedkafka/embedded-kafka/issues/202.  
Will update scala 2.12 version to 2.12.14, should be OK with the currently pulled kafka version.

**Comment by Z1kkurat** [comment]
image

File: core/src/main/scala/com/evolutiongaming/kafka/flow/RebalanceListener.scala [hunk 1]
```diff
@@ -5,46 +5,50 @@ import cats.data.NonEmptySet
...
-  ): SRebalanceListener[F] = new SRebalanceListener[F] {
+  ): SRebalanceListener1[F] = new RebalanceListener1WithConsumer[F] {
 
-    def onPartitionsAssigned(topicPartitions: NonEmptySet[TopicPartition]) =
+    override def onPartitionsAssigned(topicPartitions: NonEmptySet[TopicPartition]): RebalanceCallback[F, Unit] =
       groupByTopic(topicPartitions) traverse_ { case (topic, flow, partitions) =>
         for {
-          log <- log[F]
-          _ <- log.prefixed(topic).info(s"$partitions assigned")
+          log <- log[F].lift
```
Comment (t3hnar) [inline core/src/main/scala/com/evolutiongaming/kafka/flow/RebalanceListener.scala:26]: please minimise number of `lift` calls

Comment (t3hnar) [reply core/src/main/scala/com/evolutiongaming/kafka/flow/RebalanceListener.scala:26]: like one before  `consumer.position` and one after

Comment (Z1kkurat) [reply core/src/main/scala/com/evolutiongaming/kafka/flow/RebalanceListener.scala]: done

**Pull request merged**